### PR TITLE
Finish adding proper support for ServerDL and FastDL

### PR DESCRIPTION
**Pull request by Xalalau**
Details are in the commit comments.

### CODE REVIEW

File: lua/downloader/legacy_scanner.lua [hunk 1]
```diff
@@ -28,6 +28,13 @@ local function ScanAddons()
     local currentMap = game.GetMap()
 
     local legacyFiles = {}
+    local isFastDL = GetConVar("sv_downloadurl"):GetString() ~= "" and true
```
Comment (ceifa) [inline lua/downloader/legacy_scanner.lua]: and true?

File: lua/downloader/legacy_scanner.lua [hunk 2]
```diff
@@ -28,6 +28,13 @@ local function ScanAddons()
...
+    local isServerDL = not isFastDL
+
+    if isServerDL and not GetConVar("sv_allowdownload"):GetBool() then
+        print("[DOWNLOADER] ERROR! YOU ARE TRYING TO USE SERVERDL WITH 'sv_allowdownload' SET TO 0! CHANGE IT TO 1 BEFORE SCANNING. SKIPING STEP...")
```
Comment (ceifa) [inline lua/downloader/legacy_scanner.lua]: Use ErrorNoHalt instead

File: lua/downloader/legacy_scanner.lua (hunk 1 above)
Comment (Xalalau) [reply lua/downloader/legacy_scanner.lua]: hu3

File: lua/downloader/legacy_scanner.lua (hunk 2 above)
Comment (Xalalau) [reply lua/downloader/legacy_scanner.lua]: img
Nice

**Pull request merged**